tests/test_k3_per_loop_semaphore.py

```
@@ -0,0 +1,63 @@
+    release_response = threading.Event()
+
+    class GatewayHandler(BaseHTTPRequestHandler):
+        def do_GET(self) -> None:
+            request_started.set()
+            release_response.wait()
+            self.send_response(200)
+            self.send_header("Content-Length", str(len(expected_body)))
+            self.end_headers()
+            self.wfile.write(expected_body)
+
+        def log_message(self, format: str, *args: object) -> None:
```

Ruff (0.15.20)

[error] Function argument `format` is shadowing a Python builtin

(A002)